vmss-prototype/vmss-prototype

```diff
@@ -952,6 +953,16 @@ def NodeName(v):
         raise argparse.ArgumentTypeError("Node name must be a vmss node:  eg k8s-pool-131414-vmss0003ax")
     return v
 
+def NewUpdatedNodes(v):
+    """
+    new-updated-nodes validation type for argparser
+    :param v: The proposed new-updated-nodes value
+    :return: The validated new-updated-nodes
+    """
+    if not isinstance(v, int):
+        raise argparse.ArgumentTypeError("new-updated-nodes must be an int:  eg 1")
```

Comment:
I would expect that you not only want it to be an int but one not less than 0 :-)

Setting this to -1 would not be good, for example.

Reply:
Oops, just noticed you checked for >0 so negative would be the same as zero.  Still, maybe it should be an error rather than collapsing into 0.

Reply:
Agree that's what input validation is for, updated